contracts/Core/BlockManager.sol

```diff
@@ -245,6 +224,31 @@ contract BlockManager is Initializable, ACL, BlockStorage, StateManager, IBlockM
         return true;
     }
 
+    function _confirmBlock(
+        uint32 epoch,
+        uint8[] memory deactivatedAssets,
+        uint32 stakerId
+    ) internal {
+        uint8 blockId = sortedProposedBlockIds[epoch][uint8(blockIndexToBeConfirmed)];
+        for (uint8 i = uint8(deactivatedAssets.length); i > 0; i--) {
+            // slither-disable-next-line calls-loop
+            uint8 index = assetManager.getAssetIndex(deactivatedAssets[i - 1]);
```

Comment:
possible to get asset index array in one call before loop, instead of calling this in every iteration?

Reply:
That isn't possible with how getAssetIndex works currently.